Examples/ShareSavedSearch.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+
+namespace IReportsApiExamples.Examples
+{
+    public class ShareSavedSearch
+    {
+        /// <summary>Searches through all of a specific user's saved searches for any search with the desired search terms.
+        /// If one already exists, it makes it shared. If not, it creates a new saved search with the desired search terms
+        /// and sets the title as the search term and makes it shared</summary>
+
+        /// <param name="iReportsLibrary">The IReportsLibrary object to use</param>
+        /// <param name="username">The username of the desired user</param>
+        /// <param name="searchTerms">The desired search terms to search for in all the saved searches</param>
+        public static async Task DoWork(IReportsLibrary iReportsLibrary, string username,
+                                        string searchTerms)
+        {
+            ObservableCollection<SavedSearchListModel> savedSearchesList
+                        = await iReportsLibrary.GetSavedSearchesAsync(username);
+
+            SavedSearchModel savedSearchModel = await FindSavedSearch(iReportsLibrary, savedSearchesList, username, searchTerms);
+
+            if (savedSearchModel == null)
+            {
+                await iReportsLibrary.PostSavedSearchAsync(
+                    username, CreateNewSavedSearchWithSpecifiedSearchTerms(searchTerms));
+
+                Console.WriteLine(
+                    $@"Created new Saved Search with a title of '{searchTerms}'
+                    which is shared and has the specified search terms");
```

Comment:
If you just return here then you don't need the else statement. This means that it is more expandable if you want to add other state checks